SeafShare/ShareViewController.m

```diff
@@ -0,0 +1,283 @@
+//
+//  ShareViewController.m
+//  SeafShare
+//
+//  Created by three on 2018/7/26.
+//  Copyright © 2018年 Seafile. All rights reserved.
+//
+
+#import "ShareViewController.h"
+#import "UIViewController+Extend.h"
+#import "SeafGlobal.h"
+#import "SeafStorage.h"
+#import "Debug.h"
+#import "Utils.h"
+#import "SeafUploadFile.h"
+#import "SeafCell.h"
+#import "SeafDateFormatter.h"
+#import "FileSizeFormatter.h"
+#import "SeafShareAccountViewController.h"
+#import "SeafShareDirViewController.h"
+#import "SeafDataTaskManager.h"
+#import "SeafInputItemsProvider.h"
+
+@interface ShareViewController ()<UITableViewDataSource, UITableViewDelegate, SeafUploadDelegate>
+
+@property (nonatomic, copy) NSArray *conns;
+@property (nonatomic, strong) NSArray *ufiles;
+@property (nonatomic, strong) SeafConnection *connection;
+@property (nonatomic, strong) SeafDir *directory;
+@property (nonatomic, strong) UIActivityIndicatorView *loadingView;
+@property (nonatomic, strong) UIAlertController *alert;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (weak, nonatomic) IBOutlet UILabel *accountLabel;
+@property (weak, nonatomic) IBOutlet UILabel *destinationLabel;
+@property (weak, nonatomic) IBOutlet UIButton *accontButton;
+@property (weak, nonatomic) IBOutlet UIButton *destinationButton;
+@property (weak, nonatomic) IBOutlet UIButton *saveButton;
+
+@end
+
+@implementation ShareViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    if (IsIpad()) {
+        [self setPreferredContentSize:CGSizeMake(480.0f, 540.0f)];
+    }
+    
+    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:self action:@selector(cancel:)];
+    self.navigationItem.title = NSLocalizedString(@"Save to Seafile", @"Seafile");
+    
+    self.accountLabel.text = NSLocalizedString(@"Accounts", @"Seafile");
+    self.destinationLabel.text = NSLocalizedString(@"Destination", @"Seafile");
+    self.accontButton.enabled = false;
+    self.destinationButton.enabled = false;
+    [self.saveButton setTitle:NSLocalizedString(@"Save", @"Seafile") forState:UIControlStateNormal];
+    
+    [SeafGlobal.sharedObject loadAccounts];
+    _conns = SeafGlobal.sharedObject.conns;
+    if (_conns.count > 0) {
+        self.connection = _conns.firstObject;
+        NSString *hostAndName = [NSString stringWithFormat:@"%@-%@",_connection.host,_connection.username];
+        [self.accontButton setTitle:hostAndName forState:UIControlStateNormal];
+        [self updateSaveButton];
+    } else {
+        [self showAlert];
+    }
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(selectDirNotification:) name:@"SelectedDirectoryNotif" object:nil];
+    
+    [self setupTableview];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
```

Comment:
Do not do it inside viewWillAppear, it will be called every time user gets into the UI,